task_opts.go

```diff
@@ -36,11 +36,22 @@ func WithRootFS(mounts []mount.Mount) NewTaskOpts {
 	}
 }
 
+func getCheckpointOptions(r *CheckpointTaskInfo) *runctypes.CheckpointOptions {
+	var opts *runctypes.CheckpointOptions
+	if r.Options == nil {
+		opts = &runctypes.CheckpointOptions{}
```

Comment:
So, why are we using `interface{}` type for `CheckpointTaskInfo.Options` field? Wouldn't it be cleaner for it to be of `runctypes.CheckpointOptions`? I.e. have
```go

// TaskInfo sets options for task creation
type TaskInfo struct {
        // Checkpoint is the Descriptor for an existing checkpoint that can be used
        // to restore a task's runtime and memory state
        Checkpoint *types.Descriptor
        // RootFS is a list of mounts to use as the task's root filesystem
        RootFS []mount.Mount
        // Options hold runtime specific settings for task creation
        Options runctypes.CheckpointOptions
}
```

At least this way we'll avoid doing tricks like the one in here, as well as typecasting it every time.